src/server/main.cpp

```diff
@@ -14,16 +14,54 @@
 
 #include <memory>
 #include <thread>
+#include <queue> 
+#include <mutex> 
+#include <condition_variable>
 
 #include <iostream>
 namespace
 {
 
 std::unique_ptr<alicia::LoginDirector> g_loginDirector;
 std::unique_ptr<alicia::RanchDirector> g_ranchDirector;
+std::queue<std::function<void()>> g_taskQueue; // Task queue to hold lambdas.
+std::mutex g_queueMutex;                      // Mutex for queue access.
+std::condition_variable g_queueCondition;     // For notifying the server thread.
 
 } // namespace
 
+// Adds task to queue
+void EnqueueTask(std::function<void()> task)
+{
+    {
+        std::lock_guard<std::mutex> lock(g_queueMutex);
+        g_taskQueue.push(std::move(task));
+    }
+    g_queueCondition.notify_one();
+}
+
+// Processes tasks
+void ProcessTasks()
+{
+    while (true) 
+    {
+        std::function<void()> task;
+
+        {
+            std::unique_lock<std::mutex> lock(g_queueMutex);
+            g_queueCondition.wait_for(lock, std::chrono::milliseconds(50), [] { return !g_taskQueue.empty(); });
```

Comment:
Why wait 50ms for each task?

Reply:
I just wanted to make sure each task gets enough processing time but it seems like it's not needed. I'll get rid of it